src/management-system-v2/components/auth.tsx

```diff
@@ -18,20 +19,43 @@ export const getCurrentUser = cache(async () => {
 // be called when the session is first accessed and everything above can PPR. For
 // permissions, each server component should check its permissions anyway, for
 // composability.
-export const getCurrentEnvironment = cache(async (activeEnvironment?: string) => {
-  const { userId } = await getCurrentUser();
+type PermissionErrorHandling =
+  | { action: 'redirect'; redirectUrl?: string }
+  | { action: 'throw-error' };
+export const getCurrentEnvironment = cache(
+  async (
+    activeEnvironment?: string,
+    _opts?: { permissionErrorHandling: PermissionErrorHandling },
+  ) => {
+    const opts = _opts || { permissionErrorHandling: { action: 'redirect' } };
```

Comment:
You can provide a default value if the arg is undefined:

```suggestion
    opts: { permissionErrorHandling: PermissionErrorHandling } = { permissionErrorHandling: { action: 'redirect' } },
  ) => {
```